bridge/src/main/java/io/stargate/bridge/service/BridgeService.java

```diff
@@ -88,7 +107,37 @@ public void executeBatch(Batch batch, StreamObserver<Response> responseObserver)
   public void describeKeyspace(
       Schema.DescribeKeyspaceQuery request,
       StreamObserver<Schema.CqlKeyspaceDescribe> responseObserver) {
-    SchemaHandler.describeKeyspace(request, persistence, responseObserver);
+
+    String decoratedName =
+        persistence.decorateKeyspaceName(request.getKeyspaceName(), GrpcService.HEADERS_KEY.get());
+
+    Keyspace keyspace = persistence.schema().keyspace(decoratedName);
+    if (keyspace == null) {
+      responseObserver.onError(
+          Status.NOT_FOUND.withDescription("Keyspace not found").asException());
+    } else if (request.hasHash() && request.getHash().getValue() == keyspace.hashCode()) {
+      // Client already has the latest version, don't resend
+      responseObserver.onNext(EMPTY_KEYSPACE_DESCRIPTION);
+      responseObserver.onCompleted();
+    } else {
+      try {
+        Schema.CqlKeyspaceDescribe description = descriptionCache.get(keyspace);
```

Comment:
OK let me test this wrt perf.. I guess you are aware that this is not being updated on schema changes..